### PR TITLE
feat(Firebase): Update packages and login flow for Firebase 3

### PR DESCRIPTION
**Pull request by jemmyw**

### CODE REVIEW

File: src/components/Login/index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  return {
+    DOM: combineDOMsToDiv('.buttons', google, fb),
+    auth$: $.merge(google.auth$, fb.auth$),
+  }
+}
+
+const Login = sources => {
+  const buttons = LoginButtons(sources)
+
+  const DOM = buttons.DOM.map(buttons =>
```
Comment (TylorS) [inline src/components/Login/index.js]: I would rename one of the variables `buttons` as it is easy to confuse the 2 instances

File: src/components/SwitchedComponent.js [hunk 2]
```diff
@@ -1,19 +1,29 @@
 import {pluckLatest, pluckLatestOrNever} from 'util'
-import isolate from '@cycle/isolate'
 
-const SwitchedComponent = sources => {
+const sinks = [
```
Comment (TylorS) [inline src/components/SwitchedComponent.js:3]: :+1:

File: src/helpers/auth/index.js [hunk 3]
```diff
@@ -0,0 +1,233 @@
+/**
```
Comment (TylorS) [inline src/helpers/auth/index.js:1]: +1 to this file. However, I'm curious if by convention these should be PascalCase like they are?

PascalCase is generally reserve for Components. While this are function which take a Component and return a Component. For instance `isolate()` is not a capitalized function and it shares the same signature

``` haskell
utility :: Component -> Component
```

File: src/main.js [hunk 4]
```diff
@@ -21,7 +23,16 @@ import Root from './root'
 const history = supportsHistory() ?
   createHistory() : createHashHistory()
 
-const fbRoot = new Firebase(__FIREBASE_HOST__) // eslint-disable-line
+try {
+  firebase.app()
```
Comment (TylorS) [inline src/main.js:27]: In what case would firebase already be running?

File: src/root/index.js [hunk 5]
```diff
@@ -249,4 +155,11 @@ const IsMobile = Component => sources => {
   })
 }
 
-export default IsMobile(Root)
+// The returned component is Root wrapped in each of these helpers
```
Comment (TylorS) [inline src/root/index.js:160]: Nice!

**Comment by TylorS** [comment]
Very nice PR. Only a few comments. Thanks for taking the time to remove all the useless .tap()s and .subscribe()s :+1:

File: src/main.js (hunk 4 above)
Comment (jemmyw) [inline src/main.js:27]: when a hot reload occurs

File: src/helpers/auth/index.js (hunk 3 above)
Comment (jemmyw) [inline src/helpers/auth/index.js:1]: yes I think I agree, but I'm not sure it's worth changing here for the moment.

File: src/components/Login/index.js (hunk 1 above)
Comment (jemmyw) [inline src/components/Login/index.js]: done

File: src/helpers/auth/index.js (hunk 3 above)
Comment (TylorS) [inline src/helpers/auth/index.js:1]: yeah is suppose I agree with that as well.

**Pull request merged**